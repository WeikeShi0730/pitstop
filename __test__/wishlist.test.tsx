import { render, RenderResult } from "@testing-library/react";
import Wishlist from "../components/wishlist.component";

let documentBody: RenderResult;
describe("<Wishlist />", () => {
  it("shows empty in <Wishlist />", () => {
    documentBody = render(<Wishlist wishlistItems={[]} />);
    expect(
      documentBody.getByText("Your wishlist is empty!")
    ).toBeInTheDocument();
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
  it("shows content in <Wishlist />", () => {
    documentBody = render(
      <Wishlist
        wishlistItems={[
          {
            imageUrl: "test img",
            name: "test name",
            price: 0,
            id: "test id",
            featured: false,
            sold: 0,
            teamId: "test team id",
          },
        ]}
      />
    );
    expect(documentBody.getByText("My wishlist")).toBeInTheDocument();
    expect(documentBody.getByText("test name")).toBeInTheDocument();
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
