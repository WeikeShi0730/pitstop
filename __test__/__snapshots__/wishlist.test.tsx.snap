// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wishlist /> shows content in <Wishlist /> 1`] = `
<body>
  <div>
    <div
      class="flex flex-col h-auto lg:h-full w-full justify-start"
    >
      <div
        class="h-auto grid grid-cols-1 lg:grid-cols-2 2xl:grid-cols-3 gap-10 justify-items-center m-10"
      >
        <p
          class="flex gap-x-2 justify-self-start text-slate-700 text-lg md:text-2xl col-span-1 lg:col-span-2 2xl:col-span-3 py-1 border-b border-slate-700"
        >
          My wishlist
        </p>
        <div
          class="w-80"
        >
          <div
            class="p-5 rounded-lg text-slate-700 bg-opacity-30 bg-slate-100 transition-all duration-200 ease-in-out shadow-md hover:shadow-slate-500 text-sm md:text-base"
          >
            <div
              class="flex relative w-full h-56 justify-center items-center m-auto p-3"
            >
              <div
                class="relative w-full h-full bg-[#F8F8F8] rounded-lg shadow-md transition-all duration-200 ease-in-out hover:shadow-slate-500"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="test name image"
                    class="object-contain"
                    data-nimg="fill"
                    decoding="async"
                    src="test img"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </span>
              </div>
            </div>
            <div
              class="flex flex-col justify-center relative m-3 py-2 border-y-2 border-slate-50"
            >
              <div
                class="font-normal text-base md:text-xl my-1 text-left h-8 md:h-12"
              >
                test name
              </div>
              <div
                class="text my-1 text-left text-lg md:text-2xl"
              >
                <button>
                  â™¡
                </button>
              </div>
              <div
                class="flex items-center justify-between my-1 gap-3"
              >
                <div
                  class="flex items-end gap-1"
                >
                  <div
                    class="text-sm md:text-base"
                  >
                    CAD
                  </div>
                  <div
                    class="font-normal text-base md:text-xl"
                  >
                    0
                  </div>
                </div>
                <button
                  class="bg-orange-theme font-light text-slate-100 w-fit p-3 rounded-lg hover:bg-orange-500 hover:text-white transition-all duration-200 ease-in-out hover:shadow-md hover:shadow-orange-600"
                >
                  Add to cart
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-center m-10 mt-auto"
      >
        <div>
          <div
            aria-orientation="horizontal"
            class="flex w-fit max-w-full p-1 gap-x-1 bg-slate-300 rounded-lg shadow text-sm md:text-base"
            role="tablist"
          >
            <button
              class="buttonPrev w-8 py-1 font-light leading-5 text-slate-700 rounded-lg hover:bg-slate-100 text-slate-400 hover:bg-transparent"
              disabled=""
            >
              &lt;
            </button>
            <button
              aria-selected="true"
              class="font-light w-8 py-1 leading-5 text-slate-700 rounded-lg focus:outline-none transition-all ease-in-out duration-200 bg-orange-theme shadow text-slate-100"
              id="headlessui-tabs-tab-1"
              role="tab"
              tabindex="0"
              type="button"
            >
              1
            </button>
            <button
              class="buttonNext w-8 py-1 font-light leading-5 text-slate-700 rounded-lg hover:bg-slate-100 text-slate-400 hover:bg-transparent"
              disabled=""
            >
              &gt;
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<Wishlist /> shows empty in <Wishlist /> 1`] = `
<body>
  <div>
    <div
      class="flex flex-col h-auto lg:h-full w-full justify-start"
    >
      <div
        class="h-full grid grid-cols-1 lg:grid-cols-2 2xl:grid-cols-3 gap-10 justify-items-center m-10"
      >
        <div
          class="flex flex-grow justify-center items-center h-full col-span-1 lg:col-span-2 2xl:col-span-3"
        >
          <p
            class="px-5 gap-x-2 text-lg md:text-2xl"
          >
            Your wishlist is empty!
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
